src/v/redpanda/admin/security.cc

```diff
@@ -580,6 +582,38 @@ admin_server::oidc_revoke_handler(std::unique_ptr<ss::http::request>) {
     co_return ss::json::json_return_type(ss::json::json_void());
 }
 
+ss::future<ss::json::json_return_type> admin_server::list_user_roles_handler(
+  std::unique_ptr<ss::http::request> req, request_auth_result auth_result) {
+    ss::sstring filter = req->query_parameters["filter"];
+
+    auto member = [&auth_result]() -> std::optional<security::role_member> {
+        if (!auth_result.is_authenticated()) {
+            return std::nullopt;
+        }
+        return security::role_member{
+          security::role_member_type::user, auth_result.get_username()};
+    }();
+
+    if (!member.has_value()) {
+        throw ss::httpd::base_exception{
+          "Bad auth result", ss::http::reply::status_type::unauthorized};
+    }
+
+    auto rng = _controller->get_role_store().local().range(
+      [&filter, member = std::move(member).value()](const auto& e) {
```

Comment:
nit: Can probably take `member` by reference rather than moving it.

Reply:
done